client/components/main/index.jsx

```diff
@@ -3,17 +3,36 @@
 /**
  * External dependencies
  */
-var React = require( 'react' ),
-	classnames = require( 'classnames' );
+import React from 'react';
+import { connect }	from 'react-redux';
+import classnames from 'classnames';
 
-module.exports = React.createClass( {
+/**
+ * Internal dependencies
+ */
+import { shouldViewBeVisible } from 'state/first-view/selectors';
+
+const Main = React.createClass( {
 	displayName: 'Main',
 
 	render: function() {
+		const classes = classnames( this.props.className, 'main', {
+			'is-first-view-visible': this.props.isFirstViewVisible
+		} );
+
 		return (
-			<main className={ classnames( this.props.className, 'main' ) } role="main">
+			<main className={ classes } role="main">
 				{ this.props.children }
 			</main>
 		);
 	}
 } );
+
+export default connect(
```

Comment:
Yes, this is something we need to go through. Perhaps do this in a separate PR so people can more easily test?